sql/_35_fig_cake/cbrd_25246/cases/cbrd_25246.sql

```diff
@@ -0,0 +1,43 @@
+-- This test case verifies the following issue: CBRD-25246.
+-- Ensure that cur_pageid and volumeid columns exist and show appropriate values in the 
+-- SELECT_PAGE_INFO table when the hint /*+ select_page_info */ is used in a SELECT query
+
+drop table if exists tbl;
+create table tbl (col_a varchar(20), col_b varchar(20), col_c varchar(20), col_d varchar(20), col_e numeric(20,10), col_f varchar(20), col_g varchar(20), col_h varchar(20));
+insert into tbl select rownum, 1, mod(rownum,10), lpad(to_char(mod(rownum,100)),20,'0'), mod(rownum,1000), lpad(to_char(mod(rownum,10000)),20,'0'), lpad(to_char(mod(rownum,100000)),20,'0'), lpad(to_char(mod(rownum,100000)),20,'0') from db_class a, db_class b, db_class c, db_class d, db_class f, db_class g limit 100;
+
+--- If the test fails but all query results still show ""...PASS" then simply proceed edit the answer file to match the new results.
+WITH tbl_data AS (
+    SELECT /*+ select_page_info */ *
+    FROM tbl
+    ORDER BY p_cur_volumeid, p_cur_pageid, p_prev_pageid, p_next_pageid
+    LIMIT 2
+)
+SELECT 
+    CASE 
+        WHEN p_cur_volumeid = 0 THEN 'p_cur_volumeid=0: PASS'
+        ELSE 'p_cur_volumeid!=0: FAIL'
+    END AS assert__p_cur_volumeid,
+    CASE 
+        WHEN p_cur_pageid + 1 = p_next_pageid THEN 'p_cur_pageid+1=p_next_pageid: PASS' 
+        WHEN p_next_pageid = -1 THEN
+            CASE 
+                WHEN p_cur_pageid IS NULL THEN 'p_next_pageid=-1 & p_cur_pageid IS NULL: FAIL'
+                WHEN p_next_pageid IS NULL THEN 'p_next_pageid=-1 & p_next_pageid IS NULL: FAIL'
+                ELSE 'p_next_pageid=-1: PASS'
+            END
+        ELSE 'p_cur_pageid+1!=p_next_pageid: FAIL' 
+    END AS assert__p_cur_pageid__p_next_pageid,
+    CASE 
+        WHEN p_cur_pageid - 1 = p_prev_pageid THEN 'p_cur_pageid-1=p_prev_pageid: PASS' 
+        WHEN p_prev_pageid IS NULL THEN
+            CASE 
+                WHEN p_cur_pageid IS NULL THEN 'p_prev_pageid & p_cur_pageid IS NULL: FAIL'
+                WHEN p_next_pageid IS NULL THEN 'p_prev_pageid & p_next_pageid IS NULL: FAIL'
+                ELSE 'p_prev_pageid IS NULL: PASS'
+            END
+        ELSE 'p_cur_pageid-1!=p_prev_pageid: FAIL' 
```

Comment:
The p_cur_volumeid may not be 0. Also, the difference between the current page and the next or prev page may not be 1. What can be guaranteed is that when next is -1, cur page has the maximum value, and when prev is NULL, it has the minimum value.

Reply:
The query written here is only applicable for this test case only, meaning it is not used universally across different environments and scenarios. This means for our test case environment, the volume ID should be 0,

Reply:
As seen in the earliest commits, the answer sheets are usually hard-coded with fixed values in these scenarios; however, it was found that the starting page IDs varied for every database and may vary across future updates, which is the reason the test case was rewritten in such fashion. 
This is to say that the queries written here are applicable to this TC (i.e. the query that creates and inserts 2 pages worth of values into _tbl_) only, and are not used/reused universally across different instances and scenarios. Please correct me if I'm wrong, but unless something had gone wrong in our test case environment, the volume ID should remain 0 (as no other data exists in the testing DB) and the page numbers should be incremented by a factor of one (reflecting the inserted data in tbl after its creation).

Reply:
Adding only guaranteed test cases will prevent unnecessary errors from occurring later. But it's your job too, so do whatever you want.

Reply:
Understood. I have considered your suggestions and have made the following changes to the pass conditions in the latest commit:

- p_cur_volumeid=0  **->**  p_cur_volumeid IS NOT NULL AND p_cur_volumeid>=0
- p_cur_pageid+1=p_next_pageid  **->**  p_cur_pageid<p_next_pageid
- p_cur_pageid-1=p_prev_pageid  **->**  p_cur_pageid>p_prev_pageid